src/main/java/frc/robot/subsystems/arm/ArmKinematics.java

```diff
@@ -0,0 +1,63 @@
+package frc.robot.subsystems.arm;
+
+import edu.wpi.first.math.geometry.Translation2d;
+
+/**
+ * This class contains the kinematics for the arm.
+ * All the calculations are carefully documented in the following page:
+ * https://www.societyofrobots.com/robot_arm_tutorial.shtml#inverse_kinematics
+ * There is a table of contents containing the kinematics section, and the inverse kinematics section.
+ */
+public class ArmKinematics {
+    private final double l1;
+    private final double l2;
+
+    public ArmKinematics(double l1, double l2) {
+        this.l1 = l1;
+        this.l2 = l2;
+    }
+
+    /**
+     * Forward kinematics for the arm.
+     *
+     * @param shoulderAngle the angle of the shoulder joint. [rad]
+     * @param elbowAngle the angle of the elbow joint. [rad]
+     * @return the position of the end effector. ([m, [m])
+     */
+    public Translation2d forwardKinematics(double shoulderAngle, double elbowAngle) {
+        double theta = elbowAngle + shoulderAngle - Math.PI / 2;
+        double x = l1 * Math.cos(shoulderAngle) + l2 * Math.sin(theta);
+        double y = l1 * Math.sin(shoulderAngle) + l2 * Math.cos(theta);
+        return new Translation2d(x, y);
+    }
+
+    /**
+     * Inverse kinematics for the arm.
+     *
+     * @param x the x coordinate of the end effector. [m]
+     * @param y the y coordinate of the end effector. [m]
+     * @return the angles of the shoulder and elbow joints. ([rad, rad])
+     */
+    public InverseKinematicsSolution inverseKinematics(double x, double y) {
+        double c2 = (x * x + y * y - l1 * l1 - l2 * l2) / (2 * l1 * l2);
+        double s2 = Math.sqrt(1 - c2 * c2);
+        double shoulderAngle = Math.acos(c2);
+        double theta = Math.acos((y * (l1 + l2 * c2) - x * l2 * s2) / (x * x + y * y));
+        double elbowAngle = theta - shoulderAngle + Math.PI / 2;
+
+        return new InverseKinematicsSolution(shoulderAngle, elbowAngle);
+    }
+
+    /**
+     * A class representing the solution of the inverse kinematics.
+     */
+    public static class InverseKinematicsSolution {
```

Comment:
Where? These are all the usages I've found: 
![image](https://user-images.githubusercontent.com/31829093/213876309-d3e73975-bf6a-4907-ae6e-bcae9aff0466.png)

Reply:
Here:
<img width="695" alt="image" src="https://user-images.githubusercontent.com/42340005/213876846-6560027b-39d5-44b3-a7f3-bdb20d64cdc3.png">

Reply:
Fair enough, edited my reply.

Reply:
It's still should be public, it's being accessed from another class

Reply:
Protected classes can be accessed by classes inside the same package. Outside `ArmKinematics.java` it's used only in `PrototypeArm.java`, and both of them are in the `arm` pacakge.

Reply:
So package private should be used for this, or just public in my opinion, as there shouldn't be a restriction about the inheritance of this class

Reply:
How do you make a class package-private?
I think the best solution is to write no modifier at all. That would allow the class to be accessed inside the package only and not be subclassed ([source](https://stackoverflow.com/a/215505/10015776)).
I think this is the best solution because `InverseKinematicsSolution` shouldn't be accessed from outside the `arm` package, it's part of the control logic of the arm that shouldn't be relevant to end-users.

Reply:
(Package private is writing without a modifier)

Reply:
👍🏻 That's what I thought.